src/app/app.component.ts

```diff
@@ -10,25 +16,95 @@ import { AuthService } from './auth/auth.service';
 export class AppComponent {
   title = 'altrusight';
   sidebarOpened: boolean;
+  allNonProfitsNames: string[] = [];
+
+  // searchInput = new FormControl();
+  // filteredNonProfits: Observable<string[]>;
+
 
   // TODO: issue with auth --> auth service is not instantiated until AFTER app component is
   // so we will always have auth issues on first render until fixing this
   // https://stackoverflow.com/questions/35191617/how-to-run-a-service-when-the-app-starts-in-angular-2
-  constructor(public router: Router, public authService: AuthService) {
+  constructor(public router: Router, public authService: AuthService, private nonProfitsService: NonprofitsService) {
       this.sidebarOpened = this.authService.isLoggedIn;
+      this.nonProfitsService.getNonprofit('')
+
+      // KEEP CODE BELOW: In case we want to improve search functinality
+      // as a drop down -->
+      // Initialize array that will hold nonprofit names
+      // this.initializeNonProfitsNamesArr();
+      // this.filteredNonProfits = this.searchInput.valueChanges.pipe(
+      //   // startWith(''),
+      //   map(value => this._filter(value))
+      // );
+  }
+
+  //========================================================
+  // Helper function to filer nonprofits as user types. 
+  // This function wil trigger on user input
+  //========================================================
+  private _filter(searchParam: string): string[] {
+    if(searchParam.length != 0) {
+
+      const filterValue = searchParam.toLowerCase();
+      return this.allNonProfitsNames.filter(option => option.toLowerCase().indexOf(filterValue) >= 0);    
+    }
+
+    return [];
+  }
+
+  //========================================================
+  // Will populate nonprofits names array with results of API
+  // call.
+  // TODO: API call returns only 100 nonprofits (in alphabetical order).
+  // Need to find a way to make this better. 
+  //========================================================
+  private initializeNonProfitsNamesArr(): void {
```

Comment:
[SUGGESTION] I would perhaps remove the nonprofit names array altogether. Since we're already pulling in an array of nonprofits, might as well take advantage of what we have already and just loop through those nonprofits and just use nonprofit.charityName in the display instead of storing all the names in an array and then looping through the names to display the names.